File: server/src/main/java/com/talkeasy/server/dto/user/FollowResponse.java
package com.talkeasy.server.dto.user;

import com.talkeasy.server.domain.member.Follow;
import com.talkeasy.server.domain.member.Member;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@AllArgsConstructor
public class FollowResponse {

    private String userId;
    private String userName;
    private String imageUrl;
    private Integer gender; //0:남/1:여
    private Integer age;
    private String memo;
    private Boolean MainStatus;
    private Boolean locationStatus;

    public FollowResponse(Member member, Follow follow) {
        this.userId = member.getId();
        this.userName = member.getName();
        this.imageUrl = member.getImageUrl();
        this.memo = follow.getMemo();
        this.MainStatus = follow.getMainStatus();
        this.locationStatus = follow.getLocationStatus();
        this.age = member.getAge();
        this.gender = member.getGender();
    }
}

File: TalkEasy/core/data/src/main/java/com/ssafy/talkeasy/core/data/remote/datasource/follow/FollowResponse.kt
package com.ssafy.talkeasy.core.data.remote.datasource.follow

import com.google.gson.annotations.SerializedName
import com.ssafy.talkeasy.core.data.remote.datasource.base.DataToDomainMapper
import com.ssafy.talkeasy.core.domain.entity.response.Follow

data class FollowResponse(
    @SerializedName("userId")
    val userId: String,
    @SerializedName("followId")
    val followId: String,
    @SerializedName("userName")
    val userName: String,
    @SerializedName("imageUrl")
    val imageUrl: String,
    @SerializedName("memo")
    val memo: String,
    @SerializedName("mainStatus")
    val mainStatus: Boolean,
    @SerializedName("age")
    val age: Int?,
    @SerializedName("gender")
    val gender: Int?,
    @SerializedName("birthDate")
    val birthDate: String?,
    @SerializedName("locationStatus")
    val locationStatus: Boolean,
    @SerializedName("nickName")
    val nickName: String,
) : DataToDomainMapper<Follow> {

    override fun toDomainModel(): Follow =
        Follow(
            userId = userId,
            followId = followId,
            userName = userName,
            imageUrl = imageUrl,
            memo = memo,
            mainStatus = mainStatus,
            age = age,
            birthDate = birthDate,
            locationStatus = locationStatus,
            gender = gender,
            nickName = nickName
        )
}